controllers/replica_set_controller.go

```diff
@@ -155,12 +136,18 @@ type ReplicaSetReconciler struct {
 	secretWatcher    *watch.ResourceWatcher
 }
 
+// +kubebuilder:rbac:groups=mongodbcommunity.mongodb.com,resources=mongodbcommunity,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=mongodbcommunity.mongodb.com,resources=mongodbcommunity/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=mongodbcommunity.mongodb.com,resources=mongodbcommunity/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list
+
 // Reconcile reads that state of the cluster for a MongoDB object and makes changes based on the state read
 // and what is in the MongoDB.Spec
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r ReplicaSetReconciler) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
```

Comment:
nit: although the context is not used here can call still name this variable `ctx` to be explicit.

Reply:
Done!